### PR TITLE
MAINT: special: fix some pyflakes warnings

### PR DESCRIPTION
**Pull request by person142**

### CODE REVIEW

File: scipy/special/tests/test_basic.py [hunk 1]
```diff
@@ -1666,8 +1666,9 @@ def test_euler(self):
         eu0 = special.euler(0)
         eu1 = special.euler(1)
         eu2 = special.euler(2)   # just checking segfaults
-        assert_almost_equal(eu0[0],1,8)
-        assert_almost_equal(eu2[2],-1,8)
+        assert_allclose(eu0, [1], rtol=1e-15)
+        assert_allclose(eu1, [1, 0], rtol=1e-15)
```
Comment (rgommers) [inline scipy/special/tests/test_basic.py]: comparing against an expected value of ``0`` with ``atol=0`` usually gives problems. The original test had ``decimal=8`` - not sure if that was the right level, but you're tightening it a lot here. In any case should have ``atol`` not equal to zero.

**Comment by rgommers** [comment]
LGTM expect for the one test tolerance thing.

File: scipy/special/tests/test_basic.py (hunk 1 above)
Comment (person142) [inline scipy/special/tests/test_basic.py]: Note that all of the odd Euler numbers are exactly zero--i.e. as a user I'd be pretty annoyed if instead they were only zero up to (any) small `atol`. (`special.chebyt(10).coeffs` grrr...) I think this is an instance where we want literal floating point equality.

Comment (person142) [inline scipy/special/tests/test_basic.py]: Just to clarify, certainly most of the time we lose relative tolerance around zeros of functions. But in this case all we have to do is check whether the input is odd and if it is return `0`, so IMO we should get it exactly right.

Looking at the specfun [source](https://github.com/scipy/scipy/blob/master/scipy/special/specfun/specfun.f#L2560) though it appears that the odd values don't get initialized--so the fact that we're getting it right now is probably just luck. (From the f2py source it doesn't look like arrays get initialized to `0`, though I could easily have missed something.)

Comment (rgommers) [inline scipy/special/tests/test_basic.py]: > Note that all of the odd Euler numbers are exactly zero

Ah, missed that.

> But in this case all we have to do is check whether the input is odd and if it is return 0, so IMO we should get it exactly right.

Okay, that's what the changed test does. So let's put it in and see if anything falls over.

**Comment by rgommers** [comment]
Thanks Josh

**Pull request merged**